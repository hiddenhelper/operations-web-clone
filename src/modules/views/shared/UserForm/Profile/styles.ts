import { createStyles, makeStyles } from '@material-ui/core/styles';
export const useStyles = makeStyles(theme =>
  createStyles({
    inputBottomPadding: {
      paddingBottom: 25,
    },
    profileFormPaddingTop: {
      paddingTop: 22,
    },
    formButtonWrapper: {
      display: 'flex',
      justifyContent: 'flex-end',
      paddingTop: 5,
      paddingBottom: 40,
    },
    passwordButtonWrapper: {
      paddingTop: 33,
    },
    errorInputWrapper: {
      position: 'relative',
      width: '100%',
    },
    errorPosition: {
      bottom: '2px',
    },
    errorPositionProcore: {
      bottom: 'unset',
    },
  })
);
