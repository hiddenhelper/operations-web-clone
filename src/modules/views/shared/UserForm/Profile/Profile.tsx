import React, { memo } from 'react';

import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import ControlledError from '../../FormHandler/ControlledError';
import ControlledInput from '../../FormHandler/ControlledInput';
import ControlledButton from '../../FormHandler/ControlledButton';
import ControlledRadio from '../../FormHandler/ControlledRadio';
import ButtonLoader from '../../ButtonLoader';

import { UserModel, GeneralModel } from '../../../../models';
import { LANG } from '../../../../../constants';
import { IFormRules } from '../../../../../utils/useValidator';
import { getConditionalDefaultValue, getDefaultValue } from '../../../../../utils/generalUtils';
import { useStyles as buttonStyles } from '../../FormHandler/ControlledButton/styles';
import { formGlobalStyles, inputGlobalStyles } from '../../../../../assets/styles';
import { useStyles } from './styles';
import PhoneNumberInput from '../../PhoneNumberInput';

export interface IProfile {
  model: UserModel.IAccount;
  account: UserModel.IAccount;
  formRules: IFormRules;
  errors: any;
  loading: boolean;
  hasChanges: boolean;
  onSubmit: () => void;
  onDiscard: () => void;
  onChange: (model: any) => void;
  countryList?: GeneralModel.INamedEntity[];
  isFcaUser: boolean;
}

const Profile = ({ model, formRules, hasChanges, errors, loading, onDiscard, onSubmit, onChange, countryList, isFcaUser }: IProfile) => {
  const classes = useStyles();
  const buttonClasses = buttonStyles();
  const inputGlobalClasses = inputGlobalStyles();
  const formClasses = formGlobalStyles();

  return (
    <div className={classes.profileFormPaddingTop}>
      <Grid container={true}>
        <Grid item={true} xs={12} lg={12}>
          <div className={classes.errorInputWrapper}>
            <ControlledError
              show={!!errors.firstName}
              error={getConditionalDefaultValue(errors.firstName === 'is required', 'First Name is required.', errors.firstName)}
              styleClass={classes.errorPosition}
            >
              <ControlledInput label="First Name" styleClass={classes.inputBottomPadding}>
                <TextField
                  variant="outlined"
                  data-testid="profile-first-name-wrapper"
                  placeholder="First Name"
                  type="text"
                  autoComplete="off"
                  fullWidth={true}
                  name="firstName"
                  error={!!errors.firstName}
                  value={model.firstName}
                  required={formRules.firstName.required}
                  onChange={onChange}
                  inputProps={{
                    'data-testid': 'profile-first-name',
                  }}
                />
              </ControlledInput>
            </ControlledError>
          </div>
          <div className={classes.errorInputWrapper}>
            <ControlledError
              show={!!errors.lastName}
              error={getConditionalDefaultValue(errors.lastName === 'is required', 'Last Name is required.', errors.lastName)}
              styleClass={classes.errorPosition}
            >
              <ControlledInput label="Last Name" styleClass={classes.inputBottomPadding}>
                <TextField
                  variant="outlined"
                  data-testid="profile-last-name-wrapper"
                  placeholder="Last Name"
                  type="text"
                  autoComplete="off"
                  fullWidth={true}
                  name="lastName"
                  error={!!errors.lastName}
                  value={getDefaultValue(model.lastName, '')}
                  onChange={onChange}
                  inputProps={{
                    'data-testid': 'profile-last-name',
                  }}
                />
              </ControlledInput>
            </ControlledError>
          </div>
          <div className={classes.errorInputWrapper}>
            <ControlledError show={false} error={'Please enter Email.'} styleClass={classes.errorPosition}>
              <ControlledInput label="Email" styleClass={classes.inputBottomPadding}>
                <TextField
                  variant="outlined"
                  data-testid="profile-email"
                  placeholder="Email"
                  type="text"
                  autoComplete="off"
                  fullWidth={true}
                  disabled={true}
                  name="email"
                  value={getDefaultValue(model.email, '')}
                  inputProps={{
                    'data-testid': 'email',
                  }}
                />
              </ControlledInput>
            </ControlledError>
          </div>

          <div className={classes.errorInputWrapper}>
            <ControlledError
              show={!!errors.mobilePhoneNumber}
              error={getConditionalDefaultValue(errors.mobilePhoneNumber === 'is required', 'Please enter Mobile Phone Number.', errors.mobilePhoneNumber)}
              styleClass={classes.errorPosition}
            >
              <ControlledInput label="Mobile Phone Number" styleClass={`${classes.inputBottomPadding}`}>
                <PhoneNumberInput
                  countryList={countryList}
                  value={getDefaultValue(model.mobilePhoneNumber, '')}
                  onChange={onChange}
                  inputProps={{
                    variant: 'outlined',
                    error: !!errors.mobilePhoneNumber,
                    name: 'mobilePhoneNumber',
                    'data-testid': 'mobile-user-phone',
                  }}
                />
              </ControlledInput>
            </ControlledError>
          </div>
          <Grid container={true}>
            <Grid item={true} xs={9} lg={9}>
              <ControlledError
                show={!!errors.officePhoneNumber}
                error={getConditionalDefaultValue(errors.officePhoneNumber === 'is required', 'Please enter Phone Number.', errors.officePhoneNumber)}
              >
                <ControlledInput label="Office Phone Number (Optional)" styleClass={`${classes.inputBottomPadding} ${inputGlobalClasses.middleInput}`}>
                  <PhoneNumberInput
                    countryList={countryList}
                    value={getDefaultValue(model.officePhoneNumber, '')}
                    onChange={onChange}
                    inputProps={{
                      variant: 'outlined',
                      error: !!errors.officePhoneNumber,
                      name: 'officePhoneNumber',
                      'data-testid': 'office-user-phone',
                      id: 'officePhoneNumber',
                    }}
                  />
                </ControlledInput>
              </ControlledError>
            </Grid>
            <Grid item={true} xs={3} lg={3}>
              <ControlledError show={!!errors.officePhoneExtension} error={errors.officePhoneExtension}>
                <ControlledInput label="Ext" styleClass={`${classes.inputBottomPadding}`}>
                  <TextField
                    autoComplete="off"
                    variant="outlined"
                    placeholder="Ext"
                    type="text"
                    fullWidth={true}
                    name="officePhoneExtension"
                    value={getDefaultValue(model.officePhoneExtension, '')}
                    onChange={onChange}
                    error={!!errors.officePhoneExtension}
                    inputProps={{
                      'data-testid': 'office-user-phone',
                      maxLength: 6,
                    }}
                  />
                </ControlledInput>
              </ControlledError>
            </Grid>
          </Grid>
          <div className={classes.errorInputWrapper}>
            <ControlledError show={!!errors.preferredContactMethod} error={errors.preferredContactMethod}>
              <ControlledRadio
                styleClass={formClasses.radioWrapper}
                row={true}
                radioItems={[
                  {
                    value: UserModel.PreferredContactMethod.EMAIL,
                    label: LANG.EN.CONTACT_METHOD.EMAIL,
                  },
                  {
                    value: UserModel.PreferredContactMethod.PHONE,
                    label: LANG.EN.CONTACT_METHOD.PHONE,
                  },
                ]}
                formControlProps={{
                  name: 'preferredContactMethod',
                  label: 'Preferred Contact Method',
                  value: parseInt(model.preferredContactMethod as any, 10),
                  onChange: onChange,
                }}
              />
            </ControlledError>
          </div>
        </Grid>
      </Grid>
      <div className={classes.formButtonWrapper}>
        <ControlledButton styleClass={hasChanges ? buttonClasses.noOutline : ''}>
          <Button data-testid="discard-changes-btn" color="primary" className={buttonClasses.discardButton} disabled={!hasChanges} onClick={onDiscard}>
            {LANG.EN.NAVIGATION_TOP.ACTIONS.DISCARD}
          </Button>
        </ControlledButton>
        <ControlledButton styleClass={hasChanges ? buttonClasses.noOutline : ''}>
          <ButtonLoader
            text={LANG.EN.NAVIGATION_TOP.ACTIONS.SAVE}
            loadingText="Saving..."
            isLoading={loading}
            data-testid="confirm-btn"
            className={`${buttonClasses.nextButton} ${loading ? buttonClasses.loadingPadding : buttonClasses.noPadding}`}
            color="primary"
            variant="contained"
            size="large"
            onClick={onSubmit}
            disabled={!hasChanges || !model.firstName || !model.lastName || (!isFcaUser && !model.mobilePhoneNumber)}
          />
        </ControlledButton>
      </div>
    </div>
  );
};

export default memo(Profile);
