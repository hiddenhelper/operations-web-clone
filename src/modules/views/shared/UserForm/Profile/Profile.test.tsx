import React from 'react';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { render, act, fireEvent } from '@testing-library/react';

import Profile, { IProfile } from './Profile';
import { getAdminInitialState } from '../../../../../test/rootState';
import { getAccount_1 } from '../../../../../test/entities';

describe('Profile', () => {
  let wrapper;
  let props: IProfile;

  beforeEach(() => {
    props = {
      hasChanges: false,
      errors: {},
      model: getAccount_1(),
      account: getAccount_1(),
      loading: false,
      formRules: {
        firstName: { required: true },
      },
      onChangew: jest.fn(),
      onDiscard: jest.fn(),
      onSubmit: jest.fn(),
    };
  });

  it('should render', () => {
    wrapper = render(
      <Provider store={createMockStore(getAdminInitialState()) as any}>
        <Profile {...props} />
      </Provider>
    );
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should render', () => {
    wrapper = render(
      <Provider store={createMockStore(getAdminInitialState()) as any}>
        <Profile {...props} />
      </Provider>
    );
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should show validations', () => {
    wrapper = render(
      <Provider store={createMockStore(getAdminInitialState()) as any}>
        <Profile {...props} />
      </Provider>
    );
    const nameInput = wrapper.getByTestId('profile-first-name');

    act(() => {
      fireEvent.change(nameInput, { target: { name: 'firstName', value: 'test' } });
    });

    act(() => {
      fireEvent.click(wrapper.getByTestId('confirm-btn'));
    });

    act(() => {
      fireEvent.click(wrapper.getByTestId('discard-changes-btn'));
    });

    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should show validations for first name', () => {
    wrapper = render(
      <Provider store={createMockStore(getAdminInitialState()) as any}>
        <Profile {...props} />
      </Provider>
    );
    const lastNameInput = wrapper.getByTestId('profile-last-name');

    act(() => {
      fireEvent.change(lastNameInput, { target: { name: 'lastName', value: 'test' } });
    });

    act(() => {
      fireEvent.click(wrapper.getByTestId('confirm-btn'));
    });

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
