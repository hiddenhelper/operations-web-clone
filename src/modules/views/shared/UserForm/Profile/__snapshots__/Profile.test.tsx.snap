// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render 1`] = `
<body>
  <div>
    <div
      class="makeStyles-profileFormPaddingTop-2"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12"
        >
          <div
            class="makeStyles-errorInputWrapper-5"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-103 makeStyles-inputBottomPadding-1"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                First Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-first-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-first-name"
                      name="firstName"
                      placeholder="First Name"
                      required=""
                      type="text"
                      value="account name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-105 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-106"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-5"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-103 makeStyles-inputBottomPadding-1"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Last Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-last-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-last-name"
                      name="lastName"
                      placeholder="Last Name"
                      type="text"
                      value="account last name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-105 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-106"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-5"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-103 makeStyles-inputBottomPadding-1"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Email
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-email"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                      data-testid="email"
                      disabled=""
                      name="email"
                      placeholder="Email"
                      type="text"
                      value="email@test.com"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-105 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-106"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formButtonWrapper-3"
      >
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-discardButton-27 MuiButton-textPrimary Mui-disabled Mui-disabled"
            data-testid="discard-changes-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Discard Changes
            </span>
          </button>
        </span>
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained false makeStyles-nextButton-26 makeStyles-noPadding-23 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
            data-testid="confirm-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-textWrapper-111"
              >
                Save Changes
              </span>
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Profile should render 2`] = `
<body>
  <div>
    <div
      class="makeStyles-profileFormPaddingTop-113"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12"
        >
          <div
            class="makeStyles-errorInputWrapper-116"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-214 makeStyles-inputBottomPadding-112"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                First Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-first-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-first-name"
                      name="firstName"
                      placeholder="First Name"
                      required=""
                      type="text"
                      value="account name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-217"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-116"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-214 makeStyles-inputBottomPadding-112"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Last Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-last-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-last-name"
                      name="lastName"
                      placeholder="Last Name"
                      type="text"
                      value="account last name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-217"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-116"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-214 makeStyles-inputBottomPadding-112"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Email
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-email"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                      data-testid="email"
                      disabled=""
                      name="email"
                      placeholder="Email"
                      type="text"
                      value="email@test.com"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-216 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-217"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formButtonWrapper-114"
      >
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-discardButton-138 MuiButton-textPrimary Mui-disabled Mui-disabled"
            data-testid="discard-changes-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Discard Changes
            </span>
          </button>
        </span>
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained false makeStyles-nextButton-137 makeStyles-noPadding-134 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
            data-testid="confirm-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-textWrapper-222"
              >
                Save Changes
              </span>
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Profile should show validations 1`] = `
<body>
  <div>
    <div
      class="makeStyles-profileFormPaddingTop-224"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12"
        >
          <div
            class="makeStyles-errorInputWrapper-227"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-325 makeStyles-inputBottomPadding-223"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                First Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-first-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-first-name"
                      name="firstName"
                      placeholder="First Name"
                      required=""
                      type="text"
                      value="account name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-327 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-328"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-227"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-325 makeStyles-inputBottomPadding-223"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Last Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-last-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-last-name"
                      name="lastName"
                      placeholder="Last Name"
                      type="text"
                      value="account last name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-327 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-328"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-227"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-325 makeStyles-inputBottomPadding-223"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Email
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-email"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                      data-testid="email"
                      disabled=""
                      name="email"
                      placeholder="Email"
                      type="text"
                      value="email@test.com"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-327 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-328"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formButtonWrapper-225"
      >
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-discardButton-249 MuiButton-textPrimary Mui-disabled Mui-disabled"
            data-testid="discard-changes-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Discard Changes
            </span>
          </button>
        </span>
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained false makeStyles-nextButton-248 makeStyles-noPadding-245 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
            data-testid="confirm-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-textWrapper-333"
              >
                Save Changes
              </span>
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Profile should show validations for first name 1`] = `
<body>
  <div>
    <div
      class="makeStyles-profileFormPaddingTop-335"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12"
        >
          <div
            class="makeStyles-errorInputWrapper-338"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-436 makeStyles-inputBottomPadding-334"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                First Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-first-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-first-name"
                      name="firstName"
                      placeholder="First Name"
                      required=""
                      type="text"
                      value="account name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-438 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-439"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-338"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-436 makeStyles-inputBottomPadding-334"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Last Name
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-last-name-wrapper"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="profile-last-name"
                      name="lastName"
                      placeholder="Last Name"
                      type="text"
                      value="account last name"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-438 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-439"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-errorInputWrapper-338"
          >
            <div
              class="MuiFormControl-root makeStyles-formControl-436 makeStyles-inputBottomPadding-334"
            >
              <h1
                class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-alignLeft"
              >
                Email
                 
              </h1>
              <div>
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="profile-email"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                      data-testid="email"
                      disabled=""
                      name="email"
                      placeholder="Email"
                      type="text"
                      value="email@test.com"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-438 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-439"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formButtonWrapper-336"
      >
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-discardButton-360 MuiButton-textPrimary Mui-disabled Mui-disabled"
            data-testid="discard-changes-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Discard Changes
            </span>
          </button>
        </span>
        <span
          class=""
          data-testid="controlled-button"
          tabindex="-1"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained false makeStyles-nextButton-359 makeStyles-noPadding-356 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
            data-testid="confirm-btn"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-textWrapper-444"
              >
                Save Changes
              </span>
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</body>
`;
